webextension/background.js

```diff
@@ -38,37 +38,56 @@ class StudyLifeCycleHandler {
    * call `this.enableFeature` to actually do the feature/experience/ui.
    */
   constructor() {
+    /*
+     * IMPORTANT:  Listen for `onEndStudy` before calling `browser.study.setup`
+     * because:
+     * - `setup` can end with 'ineligible' due to 'allowEnroll' key in first session.
+     *
+     */
     browser.study.onEndStudy.addListener(this.handleStudyEnding);
     browser.study.onReady.addListener(this.enableFeature);
   }
 
   /**
-   * do some cleanup / 'feature reset'
+   * Cleanup
    *
    * (If you have privileged code, you might need to clean
    *  that up as well.
    * See:  https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/lifecycle.html
+   *
+   * @returns {undefined}
    */
-  async cleanup() {
-    await browser.storage.local.clear();
+  cleanup() {
+    // Commenting in favor of handling uninstall inside privileged code in cloudstorage.js
+    // This is to ensure cleanUpPrefs is triggered for all uninstall cases including
+    // manual uninstall from about:addons not currently handled by v5
+    // browser.cloudstorage.cleanUpPrefs();
   }
 
   /**
+   *
+   * side effects
    * - set up expiration alarms
    * - make feature/experience/ui with the particular variation for this user.
+   *
+   * @param {object} studyInfo browser.study.studyInfo object
+   *
+   * @returns {undefined}
    */
-  async enableFeature(studyInfo) {
+  enableFeature(studyInfo) {
     if (studyInfo.timeUntilExpire) {
-      browser.alarms.create(studyInfo.timeUntilExpire, () =>
-        browser.study.endStudy("expired"),
-      );
+      const alarmName = `${browser.runtime.id}:studyExpiration`;
```

Comment:
this change looks unrelated to the subject of the commit message...

Reply:
This sets alarm for timeleft until expiration of study. This code is getting invoked on install and when alarm goes off triggers uninstall of add-on.
This code snippet is taken from small study example of shield utils and should be reviewed there as well https://github.com/mozilla/shield-studies-addon-utils/blob/develop/examples/small-study/src/background.js#L80